Source/WebCore/page/Navigation.cpp

```diff
@@ -83,12 +85,32 @@ bool Navigation::canGoForward() const
     return true;
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#getting-the-navigation-api-entry-index
+static std::optional<size_t> getEntryIndexOfHistoryItem(Vector<Ref<NavigationHistoryEntry>>& entries, const HistoryItem& item)
+{
+    size_t index = 0;
+    for (auto& entry : entries) {
```

Comment:
Ideally we would have a better algorithm eventually rather than just searching through a Vector.